### PR TITLE
example: add example for IMU sensor range settings

### PR DESCRIPTION
**Pull request by bstbud**
### Description

example: add example for IMU sensor range settings

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [x] Functionality change
    [ ] Breaking change

### CODE REVIEW

**Comment by aliphys** [comment]
TLDR: **How can I find the units for acceleration, corresponding to senor ID 4 i.e. `SENSOR_ID_ACC`?**

---
In the context of the [IMURangeSettigns.ino](https://github.com/bstbud/nicla-sense-me-fw//blob/f9d1619646bf4722a4fdf6ad31e3fdaba7b3ca1d/Arduino_BHY2/examples/IMURangeSettings/IMURangeSettings.ino) example, I observed the following output from the Serial monitor. The Nicla Sense ME was placed still on the table, with the z-axis pointing upwards:

```
acceleration: XYZ values - X: -550   Y: -360   Z: 8219

gyroscope: XYZ values - X: -4   Y: 2   Z: 1

acceleration: XYZ values - X: -566   Y: -330   Z: 8224

gyroscope: XYZ values - X: -1   Y: 2   Z: 0

acceleration: XYZ values - X: -532   Y: -408   Z: 8171

gyroscope: XYZ values - X: -1   Y: -2   Z: -1
``` 

Considering the influence of gravity, there appears to be approximately 1g (~ $9.8m/s^2$) of force acting downwards (in the negative z-axis).

In the [BHI260AP datasheet](https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bhi260ap-ds000.pdf) on page 97, it is indicated that the default units are Earth g-s. For reference, here's a snapshot from the datasheet:

bild

Based on the information provided in the datasheet and the serial output, it seems that the sensor is measuring around 8000 gs. Or maybe there is a scale factor I have overlooked in the library?

**Comment by bstbud** [comment]
Hi @aliphys 
You are right, as you can see from the output of the sketch: Arduino_BHY2/examples/IMURangeSettings/IMURangeSettings.ino
it says the range of the accel sensor is in g (s). E.g.:
"range of accel: +/- 4g"
which says the range of the accel sensor has been set to +/- 4g, and in this case, when the device is laid flat on the table, you should see a value of about 8192. Below is a detailed explanation:

The sensor's HW has a 16bit resolution, which means the full ADC range in LSB is 2^16 = 65536 (this value does not change)
When the range is set to +/-4g, the sensor has a value swing of +4g - (-4)g = 8g, and now the ADC range of 65536LSB is used to represent 8g , so 1g is equivalent to 65536/8 = 8192LSBs. 
So when you see an output of 8192 (LSB) from the accel sensor, you know it's representing 1g which is the expected value.

**Comment by aliphys** [comment]
Thank you, @bstbud, for your detailed insights. To ensure accuracy in our documenting this example for our users:

We should reconsider the indication in the [BHI260AP datasheet](https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bhi260ap-ds000.pdf) on page 97 stating the default units as Earth g-s:

1. The .getConfiguration() method of the SensorConfig instance, providing the range of acceleration values. These values, expressed in g ($\approx 9.8m/s^2$), signify the maximum and minimum limits.

2. The default resolution of 16 bits results in a total ADC resolution of $2^{16} = 65536$. This resolution, dividing the full +/- range, leads to the observation that 1g is equivalent to 65536/8 = 8192LSBs (least significant bits).

Considering the covered full range, I assume that the assertion that 1g is equivalent to 65536/8 = 8192LSBs aligns if the 16-bit value is signed. This interpretation implies that +65536 corresponds to +4g and -65536 to -4g. However, the question remains regarding the division by 8 in this context.

I'd appreciate your insights in clearing the rationale behind dividing by 8 in the context of the signed 16-bit value and its relationship with the gravitational units. I'm trying to understand this example, any I may have overlooked something very obvious here.

File: Arduino_BHY2/examples/IMURangeSettings/IMURangeSettings.ino [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  if (millis() - printTime >= 1000) {
+    printTime = millis();
+    Serial.print(String("acceleration (raw): ") + accel.toString());
+
+    float accelX = ((float)accel.x() / 32768.0) *4;
+    float accelY = ((float)accel.y() / 32768.0) *4;
+    float accelZ = ((float)accel.z() / 32768.0) *4;
+    String accelInG = String("X: ") + String(accelX) + String(" Y: ") + String(accelY) + String(" Z: ") + String(accelZ);
+
+    Serial.println(String("acceleration (g): ") + accelInG);
```
Comment (aliphys) [inline Arduino_BHY2/examples/IMURangeSettings/IMURangeSettings.ino:70]: Added serial output to print values in units of $g = 9.8 m/s^2 $
